src/manifolds/EmbeddedManifold.jl

```diff
@@ -0,0 +1,82 @@
+"""
+    AbstractEmbeddedManifold
+
+"""
+abstract type AbstractEmbeddedManifold <: AbstractDecoratorManifold end
+
+abstract type AbstractEmbeddingType end
+
+"""
+    IsometricEmbedding <: AbstractEmbeddingType
+
+"""
+struct IsometricEmbedding <: AbstractEmbeddingType end
+
+struct EmbeddedManifold{MT <: Manifold, NT <: Manifold, ET <: AbstractEmbeddingType} <: AbstractEmbeddedManifold
+    manifold::MT
+    embedding::NT
+    type::ET
+end
+
+struct EmbeddedRetraction{R <: AbstractRetractionMethod} <: AbstractRetractionMethod
+    retractionMethod::R
+end
+struct EmbeddedInverseRetraction{IR <: AbstractInverseRetractionMethod} <: AbstractInverseRetractionMethod
+    inverseretracttionMethod::IR
+end
+
+function embedd(M::MT,p) where {MT <: AbstractEmbeddedManifold}
+    q = allocate(p)
+    return embedd!(M, q, p)
+end
+function embedd!(M, q, p)
+    error("Embedding a point $(typeof(p)) on $(typeof(M)) not yet implemented.")
+end
+
+function inner(M::MT, p, X, Y) where {MT <: AbstractIsometricallyEmbeddedManifold}
+    return inner(M.embedding,p,X,X)
+end
+
+function inverse_retract!(M::MT, X, p, q, m::EmbeddedRetraction) where {MT <: EmbeddedManifold}
+    x = allocate(q)
+    y = allocate(p)
+    embedd!(M, x, p)
+    embedd!(M, y, q)
+    retract!(M.embedding, X, x, y)
+    project_tangent!(M, X, p, X)
+    return q
+end
+
+is_default_decorator(M::EM) where {EM <: EmbeddedManifold} = is_default_embedding(M)
+
+getEmbedding(M::AbstractEmbeddedManifold) = M.embedding
```

Comment:
```suggestion
get_embedding(M::AbstractEmbeddedManifold) = M.embedding
```

Reply:
Thanks for noticing, my prototyping was too fast here I think.